### PR TITLE
change `admin restore table` syntax to `recover table` syntax.

### PR DESCRIPTION
**Pull request by crazycs520**
change `admin restore table` syntax to `restore table` syntax.

move restore table from adminStmt to ddl stmt. Because restore table is a ddl operation. 
This is for PR: https://github.com/pingcap/tidb/pull/9090

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -5782,25 +5782,22 @@ AdminStmt:
 	}
 |	"ADMIN" "RESTORE" "TABLE" "BY" "JOB" NUM
 	{
-		$$ = &ast.AdminStmt{
-			Tp: ast.AdminRestoreTable,
-			JobIDs: []int64{$6.(int64)},
+		$$ = &ast.RestoreTableStmt{
```
[the remark this answers is not in the document]
Comment (crazycs520) [reply parser.y]: PTAL again.

**Comment by kennytm** [comment]
Since you're changing the syntax anyway, could we consider whether the keyword `RESTORE` is appropriate? In most database systems `RESTORE` is typically used together with `BACKUP`, not to undo a `DROP TABLE`, e.g.

* The removed [`RESTORE TABLE`](https://mariadb.com/kb/en/library/restore-table-removed/) statement in MariaDB
* The [`RESTORE`](https://docs.microsoft.com/en-us/sql/t-sql/statements/restore-statements-transact-sql?view=sql-server-2017) statements in T-SQL
* The [`RESTORE DATABASE`](https://www.ibm.com/support/knowledgecenter/en/SSEPGG_11.1.0/com.ibm.db2.luw.admin.cmd.doc/doc/r0001976.html) command in DB2
* The [`pg_restore`](https://www.postgresql.org/docs/current/app-pgrestore.html) utility in PostgreSQL

Perhaps `RECOVER` (example reference of the term: [ApexSQL Recover](https://dba.stackexchange.com/questions/38872/how-to-undo-drop-table-statement) / [DB2](https://www.ibm.com/support/knowledgecenter/SSEPGG_11.1.0/com.ibm.db2.luw.admin.ha.doc/doc/t0006318.html) / [TwinDB](https://www.quora.com/How-ro-Recover-dropped-table-from-MYSQL-database)) would be better. Additionally [Oracle](https://docs.oracle.com/cd/B19306_01/backup.102/b14192/flashptr004.htm) names the command `FLASHBACK` but still uses the verb "recover" in the docs.

File: parser.y [hunk 2]
```diff
@@ -622,6 +621,7 @@ import (
 	SelectStmt			"SELECT statement"
 	RenameTableStmt         	"rename table statement"
 	ReplaceIntoStmt			"REPLACE INTO statement"
+	RecoverTableStmt                 "recover table statement"
```
Comment (zimulala) [inline parser.y]: Please align.

Comment (crazycs520) [reply parser.y]: done

**Comment by tiancaiamao** [comment]
LGTM
It seems this PR should be merge with https://github.com/pingcap/tidb/pull/9090 together

**Pull request merged**